src/points/saga.ts

```diff
@@ -66,10 +75,45 @@ export function* getHistory({ payload: params }: ReturnType<typeof getHistorySta
   }
 }
 
+export function* getPointsConfig() {
+  try {
+    const response = yield* call(fetchWithTimeout, networkConfig.getPointsConfigUrl, {
+      method: 'GET',
+    })
+    if (!response.ok) {
+      throw new Error(`Failed to fetch points config: ${response.status} ${response.statusText}`)
+    }
+    const { config }: { config: PointsConfig } = yield* call([response, 'json'])
+
+    // remove any activities that are not supported by app and have no points value
+    const supportedActivities: PointsConfig = {
+      activitiesById: {},
+    }
+    Object.entries(config.activitiesById).forEach(([activityId, activityMetadata]) => {
+      if (isPointsActivity(activityId) && activityMetadata.points > 0) {
+        supportedActivities.activitiesById[activityId] = activityMetadata
+      }
+    })
+    if (Object.keys(supportedActivities.activitiesById).length === 0) {
+      throw new Error('No supported activities found in points config')
```

Comment:
(also, this will presumably result in an infinite loop of errors that retrying will never fix)

Reply:
yeah good question...i feel weird asking for a design for the points screen for when no points activities are available :/ i guess i had assumed that there would always be points activities for app versions that support points but what you're saying is that's not the case. i think that's fair, i probably overthought this scenario.

i've removed this error throwing block - the scenario where there are no supported activities can then be handled on the UI separately to an error

Reply:
yeah, i don't think this will be a common thing, but prolly good to avoid a constant loop of errors and just deal with it in the frontend as a special case